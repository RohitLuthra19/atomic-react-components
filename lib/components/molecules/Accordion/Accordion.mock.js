const defaultAccordion = {
  className: 'accordion',
  accordionHeader: 'Accordion Header',
  accordionPanelContent: 'Panel Content',
};

export default { defaultAccordion };
