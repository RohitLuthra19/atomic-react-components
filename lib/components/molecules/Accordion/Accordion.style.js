import { css } from 'styled-components';

export default css`
  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
  details {
    width: 100%;
    padding: 10px;
    background-color: #eee;
    summary {
      cursor: pointer;
      padding: 20px 0;
    }
    &:hover,
    &:active,
    &[open] {
      background-color: #ccc;
    }
  }
`;
