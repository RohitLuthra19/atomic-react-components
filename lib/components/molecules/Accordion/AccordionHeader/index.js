// @flow
export { default } from './AccordionHeader';
export { AccordionHeaderVanilla } from './AccordionHeader';
