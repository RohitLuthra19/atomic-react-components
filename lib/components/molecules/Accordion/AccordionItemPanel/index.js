// @flow
export { default } from './AccordionItemPanel';
export { AccordionItemPanelVanilla } from './AccordionItemPanel';
