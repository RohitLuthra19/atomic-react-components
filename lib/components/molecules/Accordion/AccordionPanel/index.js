// @flow
export { default } from './AccordionPanel';
export { AccordionPanelVanilla } from './AccordionPanel';
