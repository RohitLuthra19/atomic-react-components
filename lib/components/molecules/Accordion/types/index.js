// @flow
import type { Node } from 'react';

export type AccordionProps = {
  className?: string,
  children: Node,
};

export type AccordionHeaderProps = {
  className?: string,
  children: Node,
};

export type AccordionPanelProps = {
  className?: string,
  children: Node,
};
