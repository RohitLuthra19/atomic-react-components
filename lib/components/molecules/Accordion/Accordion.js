// @flow
/**
 *
 * Accordion
 *
 */
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import styles from './Accordion.style';
import AccordionHeader from './AccordionHeader';
import AccordionPanel from './AccordionPanel';

type Props = {
  className?: string,
  accordionContent?: [],
  isOpen?: boolean,
};

// eslint-disable-next-line no-unused-vars
const Accordion = ({ className, accordionContent, isOpen }: Props): Node => (
  <div className={className}>
    {accordionContent &&
      accordionContent.map((item, index) => (
        <details
          className="accordion-item"
          key={`accordion-${index.toString()}`}
          open={item.isOpen}
        >
          <AccordionHeader className={item.headerClassName || ''}>
            {item.accordionHeader}
          </AccordionHeader>
          <AccordionPanel className={item.panelClassName || ''}>
            {item.accordionBody}
          </AccordionPanel>
        </details>
      ))}
  </div>
);

Accordion.defaultProps = {
  className: '',
  accordionContent: [],
  isOpen: false,
};

export default styled(Accordion)`
  ${styles};
`;

export { Accordion as AccordionVanilla };
