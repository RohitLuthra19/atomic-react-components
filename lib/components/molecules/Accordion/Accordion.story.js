import React from 'react';
import { storiesOf } from '@storybook/react';
// eslint-disable-next-line import/named
import { defaultAccordion } from './Accordion.mock';
import Para from '../../atoms/Para';
import Heading from '../../atoms/Heading';

// Import Styled Component to showcase variations
import Accordion from '.';

storiesOf('Molecules', module)
  .addParameters({
    jest: ['Accordion', 'AccordionVanilla'],
  })
  .add('Accordion', () => (
    <Accordion {...defaultAccordion}>
      <Accordion.panel id="accordion-1">
        <Accordion.panel.header>
          <Heading HeadingType="h3" style={{ display: 'inline-block' }}>
            Heading 1
          </Heading>
        </Accordion.panel.header>
        <Para>
          The purpose of lorem ipsum is to create a natural looking block of text (sentence,
          paragraph, page, etc.) that does not distract from the layout. A practice not without
          controversy, laying out pages with meaningless filler text can be very useful when the
          focus is meant to be on design, not content.
        </Para>
      </Accordion.panel>
      <Accordion.panel>
        <Accordion.panel.header>
          <Heading HeadingType="h3" style={{ display: 'inline-block' }}>
            Heading 2
          </Heading>
        </Accordion.panel.header>
        <Para>
          The purpose of lorem ipsum is to create a natural looking block of text (sentence,
          paragraph, page, etc.) that does not distract from the layout. A practice not without
          controversy, laying out pages with meaningless filler text can be very useful when the
          focus is meant to be on design, not content.
        </Para>
      </Accordion.panel>
    </Accordion>
  ));
