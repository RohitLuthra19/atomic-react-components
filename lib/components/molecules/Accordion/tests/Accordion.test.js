import React from 'react';
import { mount } from 'enzyme';
// eslint-disable-next-line import/named
import { defaultAccordion } from '../Accordion.mock';
import Accordion from '../index';

describe('<Accordion />', () => {
  let AccordionComponent = '';

  beforeEach(() => {
    AccordionComponent = mount(<Accordion {...defaultAccordion} />);
  });

  test('accordion renders', () => {
    expect(AccordionComponent.exists()).toBe(true);
  });

  test('should render <div/>', () => {
    expect(AccordionComponent.find('div').length).toEqual(1);
  });

  test('should render correctly', () => {
    expect(AccordionComponent).toMatchSnapshot();
  });
});
