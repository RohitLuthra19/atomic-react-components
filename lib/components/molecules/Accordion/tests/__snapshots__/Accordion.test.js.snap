// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should render correctly 1`] = `
<Accordion
  accordionHeader="Accordion Header"
  accordionPanelContent="Panel Content"
  className="accordion"
>
  <StyledComponent
    accordionHeader="Accordion Header"
    accordionPanelContent="Panel Content"
    className="accordion"
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Accordion-vybsad-0",
          "isStatic": false,
          "lastClassName": "bbzLKL",
          "rules": Array [
            [Function],
            ";details{width:100%;padding:10px;background-color:#eee;summary{cursor:pointer;padding:20px 0;}.accordion-panel-content{display:none;transition:0.4s;}&:hover,&:active,&[open],&.active{background-color:#ccc;}&[open],&.active{.accordion-panel-content{display:block;}}}",
            ";",
          ],
        },
        "displayName": "Accordion",
        "foldedComponentIds": Array [],
        "panel": [Function],
        "render": [Function],
        "styledComponentId": "Accordion-vybsad-0",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <Accordion
      accordionHeader="Accordion Header"
      accordionPanelContent="Panel Content"
      className="Accordion-vybsad-0 bbzLKL accordion"
    >
      <div
        className="Accordion-vybsad-0 bbzLKL accordion"
      >
        <AccordionPanel
          className="panel"
        >
          <details
            className="accordion-panel "
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
          >
            <AccordionHeader
              ariaLabelAttr="aria-label"
              className="accordion-header"
              key=".0"
            >
              <summary
                aria-expanded={false}
                aria-label="aria-label"
                className="accordion-header"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex="0"
              >
                Accordion Header
              </summary>
            </AccordionHeader>
            <p
              key=".1"
            >
              Panel Content
            </p>
          </details>
        </AccordionPanel>
      </div>
    </Accordion>
  </StyledComponent>
</Accordion>
`;
